src/broker/attr.c

```diff
@@ -415,27 +416,24 @@ void lsattr_request_cb (flux_t *h, flux_msg_handler_t *mh,
  ** Initialization
  **/
 
-static struct flux_msg_handler_spec handlers[] = {
-    { FLUX_MSGTYPE_REQUEST, "attr.get",    getattr_request_cb, 0, NULL },
-    { FLUX_MSGTYPE_REQUEST, "attr.list",   lsattr_request_cb, 0, NULL },
-    { FLUX_MSGTYPE_REQUEST, "attr.set",    setattr_request_cb, 0, NULL },
+static const struct flux_msg_handler_spec handlers[] = {
+    { FLUX_MSGTYPE_REQUEST, "attr.get",    getattr_request_cb, FLUX_ROLE_ALL },
+    { FLUX_MSGTYPE_REQUEST, "attr.list",   lsattr_request_cb, FLUX_ROLE_ALL },
```

Comment:
That's a hard one to split out.  The code was setting the rolemask to 0, then changing it with code like
```c
flux_msg_handler_allow_rolemask (handlers[0].w, FLUX_ROLE_ALL);
```
Since `w` is gone, I thought it best to change the initial setting rather than walk the `handlers` list or whatever.